core/application/config.go

```diff
@@ -118,3 +120,24 @@ func (c ConfigAttributes) GetString(attrName string, defaultValue string) string
 	}
 	return defaultValue
 }
+
+// GetStringMap gets the specified map attribute as map[string]string.
+func (c ConfigAttributes) GetStringMap(attrName string, defaultValue map[string]string) map[string]string {
+	if valData, ok := c[attrName]; ok {
+		result := make(map[string]string)
+		switch val := valData.(type) {
+		case map[string]string:
+			for k, v := range val {
+				result[k] = v
+			}
+		case map[string]interface{}:
+			for k, v := range val {
+				result[k] = fmt.Sprintf("%v", v)
+			}
+		default:
```

Comment:
do we need return NotValid or NotSupported here?

Reply:
yeah, fair point, will return NotValid